### PR TITLE
L1Trigger/TrackTrigger: Update return type of produce to unique_ptr.

### PR DESCRIPTION
**Pull request by gartung**
Replace shared_ptr class member with function temporary unique_ptr.

### CODE REVIEW

**Comment by gartung** [comment]
@Dr15Jones

File: L1Trigger/TrackTrigger/interface/TTClusterAlgorithm_neighbor.h [hunk 1]
```diff
@@ -111,12 +110,13 @@ class ES_TTClusterAlgorithm_neighbor : public edm::ESProducer
...
-    std::shared_ptr< TTClusterAlgorithm< T > > produce( const TTClusterAlgorithmRecord & record )
+    std::unique_ptr< TTClusterAlgorithm< T > > produce( const TTClusterAlgorithmRecord & record )
     { 
       TTClusterAlgorithm< T >* TTClusterAlgo =
         new TTClusterAlgorithm_neighbor< T >( );
 
-      _theAlgo = std::shared_ptr< TTClusterAlgorithm< T > >( TTClusterAlgo );
+      std::unique_ptr< TTClusterAlgorithm< T > > _theAlgo;
+      _theAlgo = std::unique_ptr< TTClusterAlgorithm< T > >( TTClusterAlgo );
       return _theAlgo;
```
Comment (Dr15Jones) [inline L1Trigger/TrackTrigger/interface/TTClusterAlgorithm_neighbor.h]: It should be possible to replace the entire body with
```
return std::make_unique<TTClusterAlgorithm_neighbor< T >>();
```

File: L1Trigger/TrackTrigger/interface/TTClusterAlgorithm_official.h [hunk 2]
```diff
@@ -106,12 +105,13 @@ class  ES_TTClusterAlgorithm_official: public edm::ESProducer
...
-    std::shared_ptr< TTClusterAlgorithm< T > > produce( const TTClusterAlgorithmRecord & record )
+    std::unique_ptr< TTClusterAlgorithm< T > > produce( const TTClusterAlgorithmRecord & record )
     { 
       TTClusterAlgorithm< T >* TTClusterAlgo =
         new TTClusterAlgorithm_official< T >( mWidthCut );
 
-      _theAlgo = std::shared_ptr< TTClusterAlgorithm< T > >( TTClusterAlgo );
+      std::unique_ptr< TTClusterAlgorithm< T > > _theAlgo;
+      _theAlgo = std::unique_ptr< TTClusterAlgorithm< T > >( TTClusterAlgo );
       return _theAlgo;
```
Comment (Dr15Jones) [inline L1Trigger/TrackTrigger/interface/TTClusterAlgorithm_official.h]:
```
return std::make_unique<TTClusterAlgorithm_official< T >>( mWidthCut );
```

File: L1Trigger/TrackTrigger/interface/TTStubAlgorithm_cbc3.h [hunk 3]
```diff
@@ -112,7 +111,8 @@ class ES_TTStubAlgorithm_cbc3 : public edm::ESProducer
       const TrackerTopology* const theTrackerTopo = tTopoHandle.product();
 
       TTStubAlgorithm< T >* TTStubAlgo = new TTStubAlgorithm_cbc3< T >( theTrackerGeom, theTrackerTopo, mPerformZMatching2S );
-      _theAlgo = std::shared_ptr< TTStubAlgorithm< T > >( TTStubAlgo );
+      std::unique_ptr< TTStubAlgorithm< T > > _theAlgo;
+      _theAlgo = std::unique_ptr< TTStubAlgorithm< T > >( TTStubAlgo );
       return _theAlgo;
```
Comment (Dr15Jones) [inline L1Trigger/TrackTrigger/interface/TTStubAlgorithm_cbc3.h]:
```
return std::make_unique<TTStubAlgorithm_cbc3< T >>( theTrackerGeom, theTrackerTopo, mPerformZMatching2S );
```

File: L1Trigger/TrackTrigger/interface/TTStubAlgorithm_official.h [hunk 4]
```diff
@@ -172,7 +171,8 @@ class ES_TTStubAlgorithm_official : public edm::ESProducer
 									    setBarrelCut, setRingCut, setTiltedCut, setBarrelNTilt,
 									    mPerformZMatchingPS, mPerformZMatching2S);
 
-      _theAlgo = std::shared_ptr< TTStubAlgorithm< T > >( TTStubAlgo );
+      std::unique_ptr< TTStubAlgorithm< T > > _theAlgo;
+      _theAlgo = std::unique_ptr< TTStubAlgorithm< T > >( TTStubAlgo );
       return _theAlgo;
```
Comment (Dr15Jones) [inline L1Trigger/TrackTrigger/interface/TTStubAlgorithm_official.h]:
```
return std::make_unique<TTStubAlgorithm_official< T >>( theTrackerGeom, theTrackerTopo,
                                                        setBarrelCut, setRingCut, setTiltedCut, setBarrelNTilt,
                                                        mPerformZMatchingPS, mPerformZMatching2S);
```

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

**Comment by Dr15Jones** [comment]
please test

**Comment by kpedro88** [comment]
+1

**Comment by rekovic** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**